core/src/Streamly/Internal/Data/Stream/Nesting.hs

```diff
@@ -2892,6 +2941,23 @@ splitSepBySeq_ patArr (Fold fstep initial _ final) (Stream step state) =
     -- Single Pattern
     -----------------
 
+    -- TODO: Commonize the Yield part and check the performance
+    stepOuter gst (SplitOnSeqSingle0 fs st pat) = do
+        res <- step (adaptState gst) st
+        case res of
+            Yield x s -> do
+                let jump c = SplitOnSeqSingle c s pat
+                if pat == x
+                then final fs >>= yieldReinit jump
+                else do
+                    r <- fstep fs x
+                    case r of
+                        FL.Partial fs1 ->
+                            pure $ Skip $ SplitOnSeqSingle fs1 s pat
+                        FL.Done b -> yieldReinit jump b
+            Skip s -> pure $ Skip $ SplitOnSeqSingle0 fs s pat
+            Stop -> final fs >> pure Stop
```

Comment:
See splitSepBy_ . We should have the same behavior. If we have not consumed any element we should not initialize the fold. We will be perform initialization and finalization unnecessarily.